v2/tools/generator/internal/functions/kubernetes_exporter_function.go

```diff
@@ -207,12 +210,41 @@ func (d *KubernetesExporterBuilder) exportKubernetesResources(
 			sliceToClientObjectSlice),
 	}
 
-	fn.AddParameter("_", astmodel.ContextType.AsType(codeGenerationContext))
```

Comment:
Curious as to your thinking on the `err` versus `panic` pattern here, I actually think the old code was significantly easier to read.

I sorta thought that our stance was something like:
1. Error if it's possibly a user-error.
2. Panic if it's a bug.

I've seen a few panics here and there from `AsType`, but don't recall many of them being the users fault. I've definitely seen some that were due to us not handling a particular Swagger well, although I would make that argument that those are all bugs in the generator because we _should_ handle valid Swagger correctly, or have pipeline stages that protect against invalid Swagger before it ever gets to the point where we try to generate types.

Most of the other panic -> error moves were (IIRC) either earlier in the stage of things or didn't have much of an impact on readability. This I think actually has a pretty big impact on readability (due to the ubiquity of `AsType`).

Reply:
The approach I've been taking is to panic if it's a bug in the generator, error if it's caused by external factors. 

External factors includes OpenAPI/Swagger specs that we don't correctly support.

I'm driving this change based on experience. 

By it's nature, **panic** only gives a call stack - which is next to useless when the problem is specific to just one object from a specific group/version, because there's no information to tell you which type definition has the problem. Essentially, this _requires_ someone to run the generator inside an IDE in order to debug and troubleshoot. I don't see this as acceptable.

When returning an **error**, we're very good at including context, allowing the problem to be diagnosed and solved without having to resort to a debugger. In many cases, with errors, I can see exactly what needs to be addressed. 

I also this an error is a better user experience - they get told "here's what went wrong" instead of getting a very unfriendly stack dump which says nothing more than "it went bang".

Reply:
> I've definitely seen some that were due to us not handling a particular Swagger well, although I would make that argument that those are all bugs in the generator because we should handle valid Swagger correctly

I agree that they're generator bugs. 

But which is preferable? Having to first spend an hour tracking down the source (not the line number) of the panic, or getting straight into fixing the problem because you know which new definition is causing grief.

Reply:
We could trap/wrap panics at a level that injected details about the type it was on, but then we're somewhat back to the `err` at least in the method impls (though it does keep the signatures cleaner). Go doesn't support preserving callstacks after `recover` though I don't think so that's awkward too...

I'm not 100% sold on this but maybe Go forces our hand here, and we certainly have the muscle memory around `err` more than we do around wrapping panics.